engine/src/native-layer-android.cpp

```diff
@@ -141,6 +141,7 @@ bool MCNativeLayerAndroid::doPaint(MCGContextRef p_context)
 
 void MCNativeLayerAndroid::doSetViewportGeometry(const MCRectangle& p_rect)
 {
+	doSetGeometry(m_object->getrect());
```

Comment:
@montegoulding I think this object rect lookup is unneccessary as the currently set rect of a native layer is accessible as the protected MCNativeLayer::m_rect class member.